src/_imaging.c

```diff
@@ -3721,6 +3721,134 @@ static PySequenceMethods image_as_sequence = {
     (ssizessizeobjargproc)NULL, /*sq_ass_slice*/
 };
 
+/*
+Returns 0 if all of the pixels are the same, otherwise 1.
+Skips unused bytes based on the given mode.
+*/
+static int
+_compare_pixels(
+    const char *mode, const int ysize, const int linesize,
+    const UINT8 **pixels_a, const UINT8 **pixels_b
+) {
+    // Fortunately, all of the modes that have extra bytes in their pixels use four bytes for their pixels.
+    UINT32 mask = 0xffffffff;
+    if (
+        !strcmp(mode, "RGB") || !strcmp(mode, "RGBX") ||
+        !strcmp(mode, "YCbCr") || !strcmp(mode, "HSV") || !strcmp(mode, "LAB")
+    ) {
+        // These modes have three channels in four bytes,
+        // so we have to ignore the last byte.
+        mask ^= 0xff;
+    } else if (!strcmp(mode, "LA") || !strcmp(mode, "La") || !strcmp(mode, "PA")) {
+        // These modes have two channels in four bytes,
+        // so we have to ignore the middle two bytes.
+        mask = 0xff0000ff;
```

Comment:
You shouldn't need to ignore the middle two bytes, they should be equal to the first two. See https://github.com/python-pillow/Pillow/blob/3ffa8dca446ba22f3f143331e86cfef0e9caf8f2/src/libImaging/Convert.c#L234-L242 for example.

I initially wondered if you were adding this mask for speed, but looking at the next block, removing this mask would allow `memcmp` to be used, which I expect would be faster.

Reply:
"they should be equal to the first two" You'd think so, but they're not: https://github.com/Yay295/Pillow/actions/runs/5471546420/jobs/9962793523#step:8:1757

This is why I added those last two tests. Except I couldn't find a way to set this up from Python so I marked them to skip.

Reply:
I added some logging: https://github.com/Yay295/Pillow/actions/runs/5472920560/jobs/9965699214#step:8:1759

After `im.reduce()` the data is `0xffb8b8b8`, but after `im.crop().reduce()` the data is `0xff0000b8`.

Reply:
> they should be equal to the first two

They set to the same with rgb2la conversion, but this doesn't mean that they **should** be equal

Reply:
I added a new method to ImagingCore to allow directly setting the internal bytes. All of the tests work now.